src/datetimesettings.cpp

```diff
@@ -61,32 +92,51 @@ void DateTimeSettings::setDate(const QDate &date)
 {
     QDateTime newTime = QDateTime::currentDateTime();
     newTime.setDate(date);
-    m_time.setTime(newTime.toTime_t());
+    setTime(newTime.toTime_t());
 }
 
 bool DateTimeSettings::automaticTimeUpdate()
 {
-    return m_autoSystemTime == MeeGo::QmTime::AutoSystemTimeOn;
+    return m_autoSystemTime;
 }
 
 void DateTimeSettings::setAutomaticTimeUpdate(bool enable)
 {
-    m_time.setAutoSystemTime(enable ? MeeGo::QmTime::AutoSystemTimeOn : MeeGo::QmTime::AutoSystemTimeOff);
+    if (enable != m_autoSystemTime) {
+        Maemo::Timed::WallClock::Settings s;
+
+        if (enable) {
+            s.setTimeNitz();
+        } else {
+            s.setTimeManual();
+        }
+
+        if (setSettings(s)) {
+            qDebug("Auto time status set to '%s'", enable ? "on" : "off");
```

Comment:
Is there a specific reason to output the status on lines 115 and 137? There are a couple of other places where setSettings(...) is called without any debug prints. If these are development time debug prints, then I would suggest removing/commenting.